pkg/sqlcache/informer/query_generator.go

```diff
@@ -940,40 +939,54 @@ func (l *ListOptionIndexer) validateColumn(column string) error {
 }
 
 func makeGVK(gv string, k string) string {
-       ga := strings.SplitN(gv, "/", 2)
-       g := ga[0]
-       v := ga[1]
-       return fmt.Sprintf("%s_%s_%s", g, v, k)
+	ga := strings.SplitN(gv, "/", 2)
+	if len(ga) == 1 {
+		return fmt.Sprintf("%s_%s", ga[0], k)
+	}
+	g := ga[0]
+	v := ga[1]
+	return fmt.Sprintf("%s_%s_%s", g, v, k)
 }
 
-func (l *ListOptionIndexer) validateDirective(isd *sqltypes.Sort) error {
-       if !isd.IsIndirect {
-               return nil
-       }
-       fieldsList, err := l.fieldGetterForGroupName(isd.IndirectFields[0])
-       if err != nil {
-               return err
-       }
-       gvk := makeGVK(isd.IndirectFields[0], isd.IndirectFields[1])
-       found := false
-       for _, fields := range fieldsList {
-               if smartJoin(fields) == isd.IndirectFields[2] {
-                       found = true
-                       break
-               }
-       }
-       if !found {
-               return fmt.Errorf("invalid indirect field: no entry for field %s in GVK [%s]", isd.IndirectFields[2], gvk)
-       }
-       found = false
-       for _, fields := range fieldsList {
-               if smartJoin(fields) == isd.IndirectFields[3] {
-                       return nil
-               }
-       }
-       return fmt.Errorf("invalid indirect field: no entry for field %s in GVK [%s]", isd.IndirectFields[3], gvk)
+func getGVKFieldsFromIndirectFields(indirectFields []string) (g string, v string, k string) {
+	a := strings.SplitN(indirectFields[0], "/", 2)
+	if len(a) == 2 {
+		return a[0], a[1], indirectFields[1]
+	}
+	if len(a[0]) > 0 && a[0][0] == '_' {
+		a[0] = a[0][1:]
+	}
+	return "", a[0], indirectFields[1]
 }
 
+func (l *ListOptionIndexer) validateDirective(isd *sqltypes.Sort) error {
+	if !isd.IsIndirect {
+		return nil
+	}
+	if len(isd.IndirectFields) != 4 {
+		return fmt.Errorf("expected indirect sort directive to have 4 indirect fields, got %d", len(isd.IndirectFields))
+	}
+	g, v, k := getGVKFieldsFromIndirectFields(isd.IndirectFields)
+	//gvk := makeGVK(isd.IndirectFields[0], isd.IndirectFields[1])
```

Comment:
stray commented out code?

Reply:
Yes. That was a scaffold on the way to figuring out the minimal support needed to get the test code to access schemas.